### ISSUE TITLE
Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by aneroid13** [hashicorp/terraform-provider-dns#148]
Hello, 
there is problem on dns_ptr_record creation.

### Terraform Version
Terraform v0.14.5

### Affected Resource(s)
Please list the resources as a list, for example:
- dns_ptr_record

### Terraform Configuration Files

main_init.tf
```hcl
### - DNS records config - ###
provider "dns" {
  update {
    server  = local.zoneconf.AD_server
    gssapi {
        realm    = upper(local.zoneconf.AD_domain)
        username = data.vault_generic_secret.TF_ad_nix.data["login"]
        password  = data.vault_generic_secret.TF_ad_nix.data["password"]
        
    }
  }
}

module "dns-test" {
    source = "./test1"
    name = "zzz-tst1"
    default_ip_address = "10.10.10.10"
}
```
./test-srv/module_dns.tf
```hcl
locals {
    domain = "testdomain.com"
}
variable "default_ip_address" {}
variable "name" {}

resource "dns_a_record_set" "dns_a" {
      zone = "${local.domain}."
      name = var.name
      addresses = [ var.default_ip_address ]
      ttl = 900

      lifecycle {
         ignore_changes = [ ttl ]
         }
  }

resource "dns_ptr_record" "dns_ptr" {
      zone = "10.in-addr.arpa."
      name = join(".", reverse(split(".", trimprefix(var.default_ip_address, "10."))))
      ptr = "${var.name}.${local.domain}."
      ttl = 900
      
     lifecycle {
         ignore_changes = [ ttl ]
         } 
}
```

### Expected Behavior
What should have happened?

dns_a_record_set should be created.

### Actual Behavior
What actually happened?

Error: Provider produced inconsistent result after apply

When applying changes to module.dns-test.dns_ptr_record.dns_ptr, provider
"registry.terraform.io/hashicorp/dns" produced an unexpected new value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own issue tracker.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Issue not permanent. 
dns_a_record_set create after some `terraform apply` runs.

**Comment by aneroid13** [comment]
Debug log https://gist.github.com/aneroid13/f0453403077cf9a17108f4e3728a486d#file-tf_debug_dns_ptr-log

**Comment by bmhughes** [comment]
Same issues here with Terraform v1.0.1, I can hit it with all RR type resources and usually get a plugin crash with the following stacktrace.

```
Stack trace from the terraform-provider-dns_v3.2.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xaece0a]

goroutine 15 [running]:
github.com/miekg/dns.(*Msg).Insert(0xc0004a4630, 0xc00023d5b8, 0x1, 0x1)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/miekg/dns@v1.1.36-0.20210109083720-731b191cabd1/update.go:73 +0x8a
github.com/hashicorp/terraform-provider-dns/internal/provider.resourceDnsSRVRecordSetUpdate(0xc0004ae800, 0xbee5c0, 0xc00062e0b0, 0xc0004dfc80, 0xc0004a7860)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-dns/internal/provider/resource_dns_srv_record_set.go:145 +0x95a
github.com/hashicorp/terraform-provider-dns/internal/provider.resourceDnsSRVRecordSetCreate(0xc0004ae800, 0xbee5c0, 0xc00062e0b0, 0x0, 0xffffffffffffffff)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-dns/internal/provider/resource_dns_srv_record_set.go:76 +0x5b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0000bb500, 0xe298d8, 0xc000279500, 0xc0004ae800, 0xbee5c0, 0xc00062e0b0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:318 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000bb500, 0xe298d8, 0xc000279500, 0xc00049b420, 0xc0000ab1e0, 0xbee5c0, 0xc00062e0b0, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0000b27e0, 0xe298d8, 0xc000279500, 0xc00027a910, 0xc000279500, 0xced9a0, 0xc0004def00)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0005b3560, 0xe29980, 0xc000279500, 0xc00049b2d0, 0xc0005b3560, 0xc0004def60, 0xc000496ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xced9a0, 0xc0005b3560, 0xe29980, 0xc0004def60, 0xc00007d980, 0x0, 0xe29980, 0xc0004def60, 0xc0004f0000, 0x13e)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000d0380, 0xe32d18, 0xc000282a80, 0xc000482200, 0xc00031f770, 0x128bcc0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0000d0380, 0xe32d18, 0xc000282a80, 0xc000482200, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000ac1c0, 0xc0000d0380, 0xe32d18, 0xc000282a80, 0xc000482200)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-dns_v3.2.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by bflad** [comment]
This particular Terraform `Provider produced inconsistent result after apply` error with the details of `Root resource was present, but now absent.` should be covered by the changes in #260. Those changes will be released in version 3.3.0 sometime in the near future.
